packages/react-storage/src/components/StorageBrowser/__tests__/createAmplifyAuthAdapter.spec.tsx

```diff
@@ -0,0 +1,58 @@
+import { act } from 'react-test-renderer';
+import { Amplify } from 'aws-amplify';
+import { Hub } from 'aws-amplify/utils';
+import {
+  createAmplifyAuthAdapter,
+  MISSING_BUCKET_OR_REGION_ERROR,
+} from '../adapters/createAmplifyAuthAdapter';
+
+describe('createAmplifyAuthAdapter', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.spyOn(Amplify, 'getConfig').mockReturnValue({
+      Storage: {
+        S3: {
+          bucket: 'XXXXXX',
+          region: 'region',
+        },
+      },
+    });
+  });
+
+  it('should create an auth adapter with the correct properties', () => {
+    const authAdapter = createAmplifyAuthAdapter();
+
+    expect(authAdapter).toHaveProperty('getLocationCredentials');
+    expect(authAdapter).toHaveProperty('listLocations');
+    expect(authAdapter).toHaveProperty('region');
+    expect(authAdapter.region).toEqual('region');
+    expect(authAdapter).toHaveProperty('registerAuthListener');
+  });
+
+  it('should throw if no bucket or region given', () => {
+    jest.spyOn(Amplify, 'getConfig').mockReturnValue({});
+
+    expect(createAmplifyAuthAdapter).toThrow(MISSING_BUCKET_OR_REGION_ERROR);
+  });
+
+  it('should set Hub listener when registerAuthListener is called', () => {
+    const hubListenSpy = jest.spyOn(Hub, 'listen');
+
+    const authAdapter = createAmplifyAuthAdapter();
+    const onStageChange = () => {
+      /* clear location store */
+    };
```

Comment:
Is `onStageChange` called when when you complete a stage or at the start of the next?

Reply:
fixed